---
title: "RGREATE"
output: html_document
date: "2025-04-11"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```



# libs
```{r, include=FALSE}
library(GenomicRanges)
library(data.table)
library(dplyr)
library(clusterProfiler)
library(org.At.tair.db)
library(viridis)
library(patchwork)
library(ggplot2)

```


# paths
```{r, include = FALSE}
file_paths <- list(
  chg = "/Users/eylul/Desktop/JSD-Methylation/data_figs/chg.txt.gz",
  cpg = "/Users/eylul/Desktop/JSD-Methylation/data_figs/cpg.txt.gz",
  chh = "/Users/eylul/Desktop/JSD-Methylation/data_figs/chh.txt.gz"
)
```

# GO 
```{r}
run_GO <- function(gene_list, temperature) {
  if (length(gene_list) == 0) return(NULL)
  
  ego <- enrichGO(
    gene = gene_list,
    #universe = background_genes, # context spesific, passing genes
    OrgDb = org.At.tair.db,
    keyType = "TAIR",
    ont = "ALL", 
    pAdjustMethod = "BH",
    pvalueCutoff = 0.1,
    qvalueCutoff = 0.05
  )
  
  return(as.data.frame(ego))
}
```


# dot plots 

```{r}
plot_go_dotplot <- function(ego_df, temp, context, output_dir, region_class) {
  ego_result <- suppressWarnings(
    new("enrichResult", result = ego_df, pvalueCutoff = 0.1, 
        pAdjustMethod = "BH", qvalueCutoff = 0.05)
  )
  
  dotplot_file <- file.path(output_dir, paste0("GO_dotplot_", context, "_", temp, "_", region_class, ".pdf"))
  
  pdf(dotplot_file, width = 6, height = 4)
  print(dotplot(ego_result, showCategory = 20,
                title = paste("Top 20 GO terms -", context, temp, "-", region_class)) +
          scale_color_viridis())
  dev.off()
}

```

```{r}
for (context in names(file_paths)) {
  data <- read.table(file_paths[[context]], header = TRUE, sep = "\t")
  
  temperatures <- c("10C", "16C", "22C")
  region_classes <- unique(data$class)
  
  plot_list <- list()  # collect all plots for the context
  
  for (temp in temperatures) {
    jsd_col <- paste0("JSD_bit_", temp)
    
    for (region_class in region_classes) {
      data_class <- data[data$class == region_class, ]
      
      genes <- unique(data_class$nearestTSS.gene_id[!is.na(data_class[[jsd_col]])])
      genes <- genes[genes != "" & !is.na(genes)]
      
      ego_df <- run_GO(genes, temp)
      
      if (!is.null(ego_df) && nrow(ego_df) > 0) {
        ego_result <- suppressWarnings(
          new("enrichResult", result = ego_df, pvalueCutoff = 0.1, 
              pAdjustMethod = "BH", qvalueCutoff = 0.05)
        )
        
        p <- dotplot(ego_result, showCategory = 20,
                     title = paste(context, temp, region_class)) +
          scale_color_viridis()
        
        plot_list[[paste(temp, region_class, sep = "_")]] <- p
      } else {
        message(paste("No GO terms for", context, temp, region_class))
      }
    }
  }
  
  if (length(plot_list) > 0) {
    combined_plot <- wrap_plots(plot_list, ncol = 2)  
    
    output_dir <- "GO_Enrichment_Plots"
    if (!dir.exists(output_dir)) dir.create(output_dir)
    
    output_path <- file.path(output_dir, paste0("Combined_GO_", context, ".pdf"))
    
    ggsave(output_path, combined_plot, width = 10, height = 6 * ceiling(length(plot_list)/2))
  } else {
    message(paste("No GO plots available for context:", context))
  }
}


```






